### PR TITLE
implement trainingjob controller with autoscaler

### PR DESCRIPTION
**Pull request by m3ngyang**
I have implemented edl with CRD, but I think the codes need some modification. I list some `TODO`s:
1. add a `pending` status phase to indicate that a TrainingJob
2. we should use `events` to record scale history, instead of using `TrainingJobStatus.ScaleRecords` filed
3. refactor `trainingJobUpdater.go`, the logic is not that clear after implementing scaling
4. supplement experiment docs and results #22 
5. write unit test

Plz give me more advice to improve the codes.

### CODE REVIEW

File: cmd/edl/edl.go [hunk 1]
```diff
@@ -2,50 +2,120 @@ package main
...
-	kubeconfig := flag.String("kubeconfig", "", "Path to a kube config. Only required if out-of-cluster.")
-	logLevel := flag.String("log_level", "info", "Log level can be debug, info, warn, error, crit.")
+	masterURL := flag.String("master", "", "Address of a kube master.")
+	kubeConfig := flag.String("kubeconfig", "", "Path to a kube config. Only required if out-of-cluster.")
 	maxLoadDesired := flag.Float64("max_load_desired", 0.97, `Keep the cluster max resource usage around
 		this value, jobs will scale down if total request is over this level.`)
 	flag.Parse()
 
-	lvl, err := log.LvlFromString(*logLevel)
-	candy.Must(err)
```
Comment (typhoonzero) [inline cmd/edl/edl.go:24]: There are many `if err != nil { ...` blocks, so you can write a helper function like `candy.Must` to log the error and exit.

File: pkg/autoscaler_test.go [hunk 2]
```diff
@@ -1,45 +0,0 @@
-/* Copyright (c) 2016 PaddlePaddle Authors All Rights Reserve.
```
Comment (typhoonzero) [inline pkg/autoscaler_test.go:1]: why remove the tests?

Comment (m3ngyang) [reply pkg/autoscaler_test.go:1]: Done https://github.com/PaddlePaddle/edl/pull/24/files#diff-9

File: cmd/edl/edl.go (hunk 1 above)
Comment (typhoonzero) [reply cmd/edl/edl.go:24]: Are you going to update this?

Comment (m3ngyang) [reply cmd/edl/edl.go:24]: Done. @typhoonzero

**Review by typhoonzero — COMMENTED** [review]
LGTM, @Yancey1989 can you pls also take a look at this?

File: pkg/apis/paddlepaddle/v1/training_job.go [hunk 3]
```diff
@@ -26,7 +28,39 @@ func (s *TrainingJob) NeedGPU() bool {
...
+	return q.ScaledValue(resource.Mega)
+}
+
+// Fulfillment returns the fulfillment of a trainingjob
+func (s *TrainingJob) Fulfillment() float64 {
+	minInstance := s.Spec.Trainer.MinInstance
+	maxInstance := s.Spec.Trainer.MaxInstance
+
+	if minInstance == maxInstance {
+		return 1
```
Comment (Yancey0623) [inline pkg/apis/paddlepaddle/v1/training_job.go:61]: return 1.0

**Review by Yancey0623 — APPROVED** [review]
Thanks for PR, LGTM!

**Pull request merged**